modules/ROOT/pages/hl7/hl7-connector.adoc

```diff
@@ -691,9 +708,75 @@ The maps containing extension segment data are added to the basic message map in
 
 If extension segments are used in nested groups, the list containing those segments are included in the map representing that group. Extension segments are ordered by position in the lists created by the parser, and must also be ordered by position when writing.
 
+=== Sending Acknowledgments
+
+An ACK (acknowledgement) message is an HL7 message that lets you acknowledge to a message sender that your application has received a message. ACK messages are the same as writing any other HL7 message, except you set the ACK message to what was generated during the read operation as the output message under a `Data` key.
+
+For example:
+
+[source,xml,linenums]
+----
+    <hl7-edi:read config-ref="HL7_EDI__Configuration1" doc:name="HL7 EDI"/>\
+    ...
+    <dw:transform-message doc:name="Create Outgoing Message">
+      <dw:set-payload><![CDATA[%dw 1.0
+%output application/java
+---
+{
+  Name: "ACK",
+  MSH: payload.ACK.MSH,
+  Id: "ACK",
+  Data: {
+    ACK: payload.ACK
+  }
+}]]></dw:set-payload>
+    </dw:transform-message>
+    <hl7-edi:write config-ref="HL7_EDI__Configuration" messageStructure="InMessage" doc:name="ACK"/>
+    ...
+    <file:outbound-endpoint responseTimeout="10000" doc:name="File" path="output" outputPattern="ack.edi"/>
+----
+
+The generated ACK messages have MSH data set up for sending back to the sender of the original message, so you don't need to change anything in the data to do the send.
+
+If you include an ACK message schema in your configuration, that schema is used to both receive ACK messages and generate ACK messages. If you don't specify an ACK schema, the standard `hl7/v2_5_1/ACK.esl` schema is used by default.
+
+// image::ack-datasense.jpg[Default ACK message structure]
+
+=== Reading and Validating HL7 ER7 Messages
+
+To read an HL7 message:
+
+. Search the palette for HL7 EDI and drag the HL7 EDI building block into a flow.
+
+. Go to the properties view, select the connector configuration you previously created and select the Read operation.
++
+This operation reads any byte stream into the structure described by your HL7 schemas.
++
+
+
+HL7 EDI validates the message structure when it reads it in.
+Message validation includes checking the syntax and content of the MSH and all component segments of the message.
+Normally errors are logged, accumulated, and reported in the generated ACK message that is provided in the generated data structure.
+All messages whether error free or with non-fatal errors are passed on for processing as part of the output message Map.
+Errors reading the input data can throw exceptions.
+
+Error data entered in the receive data map uses the HL7Error class, a read-only JavaBean with the following properties:
+
+[%header,cols="30a,70a"]
+|===
+|Property |Description
+|`segment` |The zero-based index within the input of the segment causing the error.
```

Comment:
"Zero-based index" to be consistent with the other lines (without "the")

Remove periods for these lines.

Reply:
done